editor/tests/org.fusesource.ide.projecttemplates.tests.integration/src/main/java/org/fusesource/ide/projecttemplates/tests/preferences/StagingRepositoriesPreferencePageIT.java

```diff
@@ -0,0 +1,128 @@
+		preferenceStore.setValue(StagingRepositoriesConstants.STAGING_REPOSITORIES, testList);
+
+		// make sure they set properly
+		assertThat(preferenceStore.getBoolean(StagingRepositoriesConstants.ENABLE_STAGING_REPOSITORIES) == true);
+		assertThat(preferenceStore.getString(StagingRepositoriesConstants.STAGING_REPOSITORIES)).isEqualTo(testList);
+		
+		// open the preference dialog with the staging repositories page and get a handle to that page
+		PreferenceDialog prefsDialog =
+			PreferencesUtil.createPreferenceDialogOn(PlatformUI.getWorkbench().getActiveWorkbenchWindow().getShell(), 
+			StagingRepositoriesConstants.STAGING_REPOSITORIES_PREFERENCE_PAGE_ID, 
+			new String[] {StagingRepositoriesConstants.STAGING_REPOSITORIES_PREFERENCE_PAGE_ID}, null);
+		prefsDialog.setPreferenceStore(preferenceStore);
+		assertThat(prefsDialog).isNotNull();
+		Object page = prefsDialog.getSelectedPage();
+		assertThat(page).isNotNull();
+		assertThat(page instanceof StagingRepositoriesPreferencePage);
+		StagingRepositoriesPreferencePage stagingPage = (StagingRepositoriesPreferencePage) page;
+		
+		// make sure that the checkbox is checked and the list is what we expect it to be
+		assertThat(stagingPage.getEnableStagingRepositoriesEditor().getBooleanValue() == preferenceStore.getBoolean(StagingRepositoriesConstants.ENABLE_STAGING_REPOSITORIES));
+		assertThat(stagingPage.getStagingRepositoryListEditor().getItemList().equals(testList));
+		assertThat(stagingPage.getStagingRepositoryListEditor().isListEnabled());
+		
+		// close it
+		boolean isClosed = prefsDialog.close();
+		assertThat(isClosed == true);
+		
+		// update the enable setting to be false to test enablement of the list control
+		preferenceStore.setValue(StagingRepositoriesConstants.ENABLE_STAGING_REPOSITORIES, false);
+		
+		// reopen the dialog and grab the page
+		prefsDialog =
+				PreferencesUtil.createPreferenceDialogOn(PlatformUI.getWorkbench().getActiveWorkbenchWindow().getShell(), 
+				StagingRepositoriesConstants.STAGING_REPOSITORIES_PREFERENCE_PAGE_ID, 
+				new String[] {StagingRepositoriesConstants.STAGING_REPOSITORIES_PREFERENCE_PAGE_ID}, null);
+		prefsDialog.setPreferenceStore(preferenceStore);
+		page = prefsDialog.getSelectedPage();
+		stagingPage = (StagingRepositoriesPreferencePage) page;
+		
+		// make sure that the list control is disabled
+		assertThat(stagingPage.getEnableStagingRepositoriesEditor().getBooleanValue() == preferenceStore.getBoolean(StagingRepositoriesConstants.ENABLE_STAGING_REPOSITORIES));
+		assertThat(!stagingPage.getStagingRepositoryListEditor().isListEnabled());
+		
+		// now set things back in case there are further tests
+		StagingRepositoriesPreferenceInitializer initializer = new StagingRepositoriesPreferenceInitializer();
+		initializer.initializeDefaultPreferences();
+	}
+	
+	@Test
+	public void testNewRepositoryDialog() throws Exception {
+		PlatformUI.getWorkbench().getDisplay().asyncExec(new Runnable() {
```

Comment:
as it is run in async and there is not sync after, the test will be always sucessful